src/pages/general/login/actions/loginRequest.ts

```diff
@@ -0,0 +1,25 @@
+import { hashHistory } from 'react-router';
+import { LoginApi } from './../../../../rest-api/login';
+import { LoginCredentials } from './../../../../model/loginCredentials';
+import { loginActionEnums } from './../../../../common/actionEnums/login';
+import { LoginResponse } from './../../../../model/loginResponse';
+
+export const loginRequestStarted = (loginCredentials : LoginCredentials) => {
+  return function(dispatcher) {
+    const promise = LoginApi.login(loginCredentials);
+
+    promise.then( data => {
+      dispatcher(loginRequestCompleted(data));
+
+      if(data.succeded === true) {
+        hashHistory.push('/students');
+      }
```

Comment:
As we discusses we will update login api and add three logins (same password)

admin
student
trainer

Each of them will redirect you to admin / student or trainer, in a first iteration you can add a switch here based on rol, in a second iteratin (included in this case) we will create a separate action and we will dispatch it (just to keep simpler this action).